server/src/main/java/com/flipkart/varadhi/services/AuthZService.java

```diff
@@ -0,0 +1,121 @@
+package com.flipkart.varadhi.services;
+
+import com.flipkart.varadhi.entities.auth.IAMPolicyRequest;
+import com.flipkart.varadhi.entities.auth.ResourceType;
+import com.flipkart.varadhi.entities.auth.RoleBindingNode;
+import com.flipkart.varadhi.exceptions.IllegalArgumentException;
+import com.flipkart.varadhi.exceptions.InvalidOperationForResourceException;
+import com.flipkart.varadhi.exceptions.ResourceNotFoundException;
+import com.flipkart.varadhi.spi.db.MetaStore;
+import com.flipkart.varadhi.spi.db.RoleBindingMetaStore;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class AuthZService {
+    private final MetaStore metaStore;
+    private final RoleBindingMetaStore roleBindingMetaStore;
+
+    public AuthZService(MetaStore metaStore, RoleBindingMetaStore roleBindingMetaStore) {
+        this.metaStore = metaStore;
+        this.roleBindingMetaStore = roleBindingMetaStore;
+    }
+
+    public RoleBindingNode createRoleBindingNode(String resourceId, ResourceType resourceType) {
+        if (!isResourceValid(resourceId, resourceType)) {
+            throw new IllegalArgumentException(String.format(
+                    "Invalid resource id(%s) for resource type(%s).",
+                    resourceId,
+                    resourceType
+            ));
+        }
+        RoleBindingNode node = new RoleBindingNode(resourceId, resourceType, new HashMap<>(), 0);
+        roleBindingMetaStore.createRoleBindingNode(node);
+        return node;
+    }
+
+    public List<RoleBindingNode> getAllRoleBindingNodes() {
+        return roleBindingMetaStore.getRoleBindingNodes();
+    }
+
+    public RoleBindingNode findRoleBindingNode(ResourceType resourceType, String resourceId) {
```

Comment:
find --> search that can return empty result
get --> exact lookup that can throw entity not found